File: 1897-redistribute-characters-to-make-all-strings-equal/1897-redistribute-characters-to-make-all-strings-equal.cpp
class Solution {
public:
    bool makeEqual(vector<string>& w) {
        
        if(w.size()==1) return 1;
        
        unordered_map< char, int> m;
        
        int x=-1;
        for( string s:w){
            for( char ch:s){
                m[ch]++;
            }
        }
        
        for( auto t:m){
            
          if(t.second % w.size() !=0) return false;
        }
        return true;
    }
};

File: 1897-redistribute-characters-to-make-all-strings-equal/1897-redistribute-characters-to-make-all-strings-equal.java
class Solution {
    public boolean makeEqual(String[] w) {
        
        if( w.length==1) return true;
        int[] m=new int[26];
        Arrays.fill(m,0);
        
        
        for( String s:w){
            for( char ch:s.toCharArray()) { m[ch-'a']++; }
        }
        for( int i=0;i<26;i++){ 
            if(m[i]==0) continue;
          if( m[i] % w.length !=0) return false;
        }
        return true;
        
    }
}